document/js-api/index.bs

```diff
@@ -1120,26 +1147,142 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
-    1. Let |map| be the [=surrounding agent=]'s associated [=extern value cache=].
-    1. If a [=extern address=] |externaddr| exists such that |map|[|externaddr|] is the same as |v|,
+        1. If |null| is present,
+            1. Return [=ref.null=] |heaptype|.
+        1. Otherwise,
+            1. Throw a {{TypeError}}.
+    1. If |type| is a subtype of [=ref=] |null| [=heap-type/func=],
+        1. If |v| is an [=Exported Function=],
+            1. Let |funcaddr| be the value of |v|'s \[[FunctionAddress]] internal slot.
+            1. Let |testresult| be [=ref_test=](|heaptype|, [=ref.func=] |funcaddr|).
+            1. If |testresult| is true,
+                1. Return [=ref.func=] |funcaddr|.
+        1. Throw a {{TypeError}}.
+    1. If |heaptype| is [=extern=],
+        1. Let |map| be the [=surrounding agent=]'s associated [=extern value cache=].
+        1. If a [=extern address=] |externaddr| exists such that |map|[|externaddr|] is the same as |v|,
+            1. Return [=ref.extern=] |externaddr|.
+        1. Let [=extern address=] |externaddr| be the smallest address such that |map|[|externaddr|] [=map/exists=] is false.
+        1. [=map/Set=] |map|[|externaddr|] to |v|.
         1. Return [=ref.extern=] |externaddr|.
-    1. Let [=extern address=] |externaddr| be the smallest address such that |map|[|externaddr|] [=map/exists=] is false.
-    1. [=map/Set=] |map|[|externaddr|] to |v|.
-    1. Return [=ref.extern=] |externaddr|.
+    1. If |type| is a subtype of [=ref=] |null| [=any=],
+        1. Let |externref| be [=ToWebAssemblyValue=](|v|, [=ref=] [=extern=]).
+        1. Let |internref| be [=extern_internalize=](|externref|).
+        1. Let |testresult| be [=ref_test=](|heaptype|, |internref|).
+        1. If |testresult| is true,
+            1. Return |internref|.
+        1. Throw a {{TypeError}}.
 1. Assert: This step is not reached.
 
 </div>
 
+<h3 id="gc-exotic-objects">Garbage Collected Objects</h3>
+
+A WebAssembly struct or array is made available in JavaScript as an <dfn>Exported GC Object</dfn>.
+An [=Exported GC Object=] is an exotic object that wraps a garbage collected WebAssembly reference value.
+Most JavaScript operations on an [=Exported GC Object=] will throw an exception or return undefined.
+
+Note: These operations may be refined in the future to allow richer interactions in JavaScript with WebAssembly structs and arrays.
+
+An [=Exported GC Object=] contains an \[[ObjectAddress]] internal slot, which holds a [=object address=] relative to the [=surrounding agent=]'s [=associated store=],
+and an \[[ObjectKind]] internal slot, which holds the string value "struct" or "array".
+
+The internal methods of an [=Exported GC Object=] use the following implementations.
+
+<div algorithm>
+  The <dfn>\[[GetPrototypeOf]] internal method of an Exported GC Object</dfn> <var ignore>O</var> takes no arguments and returns null. It performs the following steps when called:
+
+    1. Return null.
```

Comment:
I updated [[GetPrototypeOf]] to return null to match the updated google doc, also see this comment: https://github.com/WebAssembly/gc/pull/355#discussion_r1111711131